raxx_elli/lib/raxx/elli/handler.ex

```diff
@@ -30,10 +30,19 @@ defmodule Raxx.Elli.Handler do
     headers = :elli_request.headers(elli_request) |> Enum.map(fn ({k, v}) ->
       {String.downcase(k), String.downcase(v)}
     end)
-    {"Host", authority} = :elli_request.headers(elli_request) |> List.keyfind("Host", 0)
-    [host, port] = String.split(authority, ":")
+
+	[host, port] = 
+	case :elli_request.headers(elli_request) |> List.keyfind("Host", 0) do
+	  {"Host", authority} ->
+		case String.split(authority, ":") do
+			[host, port] -> [host, port]
+			[host] -> [host, "80"] # we need to set the correct port some how...
```

Comment:
The reason for this block is different. The code was breaking when there were no headers.
```ex
[host, port] = 
	case :elli_request.headers(elli_request) |> List.keyfind("Host", 0) do
	  {"Host", authority} ->
		case String.split(authority, ":") do
			[host, port] -> [host, port]
			[host] -> [host, "80"] # we need to set the correct port some how...
		end
	  _ -> ["", "0"]	
	end
```

I ran https://www.htbridge.com/ssl/ and https://www.wormly.com/test_ssl against a simple raxx_elli endpoint, and had several breaks both in our code when the headers were empty, and even in the [elli](https://github.com/elli-lib/elli) code.

We can fix the raxx_elli issues, but we should refer the elli breaks to the https://github.com/elli-lib/elli team, where the active elli development seems to be going on.

log of raxx_elli errors for empty headers:
```
iex(2)> [{:req, :GET, ["abc"], [], "/abc", {1, 0}, [], "", #PID<0.418.0>,
  {:ssl,
   {:sslsocket, {:gen_tcp, #Port<0.8475>, :tls_connection, #PID<0.172.0>},
    #PID<0.554.0>}}, {Raxx.Elli.Handler, {Elliraxx, %{}}}}, {:badmatch, nil},
 [{Raxx.Elli.Handler, :normalise_request, 1,
   [file: 'lib/raxx/elli/handler.ex', line: 35]},
  {Raxx.Elli.Handler, :handle, 2, [file: 'lib/raxx/elli/handler.ex', line: 7]},
  {:elli_http, :execute_callback, 1,
   [file: 'c:/Apps/elliraxx/deps/elli/src/elli_http.erl', line: 287]},
  {:elli_http, :handle_request, 4,
   [file: 'c:/Apps/elliraxx/deps/elli/src/elli_http.erl', line: 104]},
  {:elli_http, :keepalive_loop, 5,
   [file: 'c:/Apps/elliraxx/deps/elli/src/elli_http.erl', line: 71]},
  {:proc_lib, :init_p_do_apply, 3, [file: 'proc_lib.erl', line: 247]}]]

```
log of some elli errors: (using https://github.com/elli-lib/elli)
```txt
00:29:50.078 [error] SSL: :hello: tls_handshake.erl:176:Fatal error: insufficient security - no_suitable_ciphers

iex(2)>
00:29:50.181 [error] SSL: :hello: tls_handshake.erl:176:Fatal error: insufficient security - no_suitable_ciphers

iex(2)>
00:29:50.304 [error] SSL: {:cipher, {:alert, 2, 20, {'tls_record.erl', 229}, :undefined}}: ssl_connection.erl:842:Fatal error: unexpected message

iex(2)>
00:29:50.516 [error] SSL: {:connection, {:alert, 2, 20, {'ssl_cipher.erl', 275}, :decryption_failed}}: ssl_connection.erl:842:Fatal error: unexpected message

iex(2)>
00:29:50.516 [error] Elli request (pid #PID<0.354.0>) unexpectedly crashed:
{:case_clause, {:error, {:tls_alert, 'unexpected message'}}}
```